src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/LengthConverter.cs

```diff
@@ -181,59 +174,61 @@ public override bool CanConvertTo(ITypeDescriptorContext typeDescriptorContext,
+                if (valueString.EndsWith(s_pixelUnitStrings[i], StringComparison.OrdinalIgnoreCase))
                 {
-                    strLenUnit = PixelUnitStrings[i].Length;
+                    strLenUnit = s_pixelUnitStrings[i].Length;
                     unitFactor = PixelUnitFactors[i];
                     break;
                 }
             }
 
-            //  important to substring original non-lowered string 
-            //  this allows case sensitive ToDouble below handle "NaN" and "Infinity" correctly. 
-            //  this addresses windows bug 1177408
-            valueString = valueString.Substring(0, strLen - strLenUnit);
+            // Remove pixel unit string if present, otherwise no-op
+            valueString = valueString.Slice(0, valueString.Length - strLenUnit);
 
             // FormatException errors thrown by Convert.ToDouble are pretty uninformative.
             // Throw a more meaningful error in this case that tells that we were attempting
             // to create a Length instance from a string.  This addresses windows bug 968884
             try
             {
-                double result = Convert.ToDouble(valueString, cultureInfo) * unitFactor;
-                return result;
+                return double.Parse(valueString, cultureInfo) * unitFactor;
             }
             catch (FormatException)
             {
-                throw new FormatException(SR.Format(SR.LengthFormatError, valueString));
+                throw new FormatException(SR.Format(SR.LengthFormatError, valueString.ToString()));
             }
         }
 
-        // This array contains strings for unit types 
+        // This array contains string representations for unit known types.
         // These are effectively "TypeConverter only" units.
         // They are all expressable in terms of the Pixel unit type and a conversion factor.
-        static private string[] PixelUnitStrings = { "px", "in", "cm", "pt" };
-        static private double[] PixelUnitFactors = 
-        { 
-            1.0,              // Pixel itself
-            96.0,             // Pixels per Inch
-            96.0 / 2.54,      // Pixels per Centimeter
-            96.0 / 72.0,      // Pixels per Point
-        };
+        private static readonly string[] s_pixelUnitStrings = ["px", "in", "cm", "pt"];
+
+        /// <summary> Holds the factor value representation for units specified in <see cref="s_pixelUnitStrings"/>. </summary>
+        private static ReadOnlySpan<double> PixelUnitFactors =>
+        [ 
+            1.0,              // px - Pixel itself
+            96.0,             // in - Pixels per Inch
+            96.0 / 2.54,      // cm - Pixels per Centimeter
+            96.0 / 72.0,      // pt - Pixels per Point
+        ];
+
+        /// <summary> Holds the "Auto" string representation for <see cref="double"/> conversion. </summary>
+        private static ReadOnlySpan<char> NaNValue => ['A', 'u', 't', 'o'];
```

Comment:
Same question as https://github.com/dotnet/wpf/pull/9363#discussion_r1669514041

Reply:
We don't need this, could be a literal, I'll fix that. I just went through different iterations previously and this is a leftover.

Thank you.